knative-operator/pkg/common/health_dashboard.go

```diff
@@ -0,0 +1,92 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"os"
+
+	mfc "github.com/manifestival/controller-runtime-client"
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var logh = Log.WithName("health dashboard")
+
+const ConfigManagedNamespace = "openshift-config-managed"
+
+func InstallHealthDashboard(api client.Client) error {
+	namespace, err := getOperatorNamespace()
+	if err != nil {
+		return err
+	}
+	err = api.Get(context.TODO(), client.ObjectKey{Name: ConfigManagedNamespace}, &corev1.Namespace{})
+	if apierrors.IsNotFound(err) {
+		logh.Info(fmt.Sprintf("namespace %q not found. Skipping to create dashboard.", ConfigManagedNamespace))
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get namespace %q: %w", ConfigManagedNamespace, err)
+	}
+	instance, err := getServerlessOperatorDeployment(api, namespace)
+	if err != nil {
+		return err
+	}
+	manifest, err := manifest(instance, api, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to load dashboard manifest: %w", err)
+	}
+	logh.Info("Installing dashboard")
+	if err := manifest.Apply(); err != nil {
+		return fmt.Errorf("failed to apply dashboard manifest: %w", err)
+	}
+	logh.Info("Dashboard is ready")
+	return nil
+}
+
+// manifest returns dashboard deploymnet resources manifest
+func manifest(instance *appsv1.Deployment, apiclient client.Client, namespace string) (mf.Manifest, error) {
+	manifest, err := mfc.NewManifest(manifestPath(), apiclient, mf.UseLogger(logh.WithName("mf")))
+	if err != nil {
+		return mf.Manifest{}, fmt.Errorf("failed to read dashboard manifest: %w", err)
+	}
+
+	instance.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apps",
+		Version: "v1",
+		Kind:    "Deployment",
+	})
+	instance.SetNamespace(namespace)
```

Comment:
Isn't this overkill? Don't we know the namespace and name of the deployment from environment anyway? Seems like it's only needed to set the annotations so I'd guess we don't even need to query the deployment here?

Reply:
This is how namespace is fetched, from the container's env:
```
func getOperatorNamespace() (string, error) {
	ns := os.Getenv(installedNamespaceEnvKey)
	if ns == "" {
		return "", fmt.Errorf("the environment variable %q must be set", installedNamespaceEnvKey)
	}
	return ns, nil
}
```

Reply:
You are I believe. `instance` is a deployment and that's not really needed in this case I believe.